addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js

```diff
@@ -28,19 +31,85 @@ export class FloorScreen extends Component {
         const floor = this.pos.currentFloor;
         this.state = useState({
             selectedFloorId: floor ? floor.id : null,
+            floorHeight: "100%",
+            floorWidth: "100%",
             selectedTableIds: this.getTablesSelectedByDefault(),
             isColorPicker: false,
         });
         this.floorMapRef = useRef("floor-map-ref");
+        this.floorScrollBox = useRef("floor-map-scroll");
         const ui = useState(useService("ui"));
         const mode = localStorage.getItem("floorPlanStyle");
         this.pos.floorPlanStyle = ui.isSmall || mode == "kanban" ? "kanban" : "default";
         this.map = useRef("map");
-        onMounted(() => this.pos.openCashControl());
+        onMounted(() => {
+            this.pos.openCashControl();
+        });
         onWillStart(this.onWillStart);
+        useEffect(
+            () => {
+                this.computeFloorSize();
+            },
+            () => [this.activeFloor, this.pos.floorPlanStyle]
+        );
+        useEffect(
+            () => {
+                document.addEventListener("resize", this.computeFloorSize);
+                return () => {
+                    document.removeEventListener("resize", this.computeFloorSize);
+                };
+            },
+            () => []
+        );
```

Comment:
```suggestion
        useEffect(
            () => {
                const computeFloorSize = () => this.computeFloorSize();
                document.addEventListener("resize", computeFloorSize);
                return () => {
                    document.removeEventListener("resize", computeFloorSize);
                };
            },
            () => []
        );
```
make sure the callback is properly bound to `this`.